main.tf

```diff
@@ -1,16 +1,135 @@
+module "lambda_package" {
+  source  = "matti/resource/shell"
+  version = "~>1.0.7"
 
-module "tags" {
-  source  = "rhythmictech/tags/terraform"
-  version = "1.0.0"
+  # I want to be using `npm run-script build` here,
+  # but for some reason I can't work out `node-lambda` fails in that context
+  command     = "node-lambda package -e production -x 'event.json build.env bin .github' -I lambci/lambda:build-nodejs12.x"
+  working_dir = "${path.module}/lambda"
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+  trigger = join("", values({
+    package_lock_json = filemd5("${path.module}/lambda/package-lock.json")
+    package_json      = filemd5("${path.module}/lambda/package.json")
+    app               = filemd5("${path.module}/lambda/index.js")
+  }))
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "this" {
+  name_prefix = "${var.name}-"
+  description = "Role for ${var.name} to create secret"
+
+  assume_role_policy = data.aws_iam_policy_document.assume.json
+  tags               = var.tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+locals {
+  zipfile = "${path.module}/lambda/build/lambda-ssh-key-secret-production.zip"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_basic_execution" {
+  role       = aws_iam_role.this.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "secretsmanager_read_write" {
+  role       = aws_iam_role.this.name
+  policy_arn = "arn:aws:iam::aws:policy/SecretsManagerReadWrite"
+}
+
+resource "aws_lambda_function" "this" {
+  depends_on = [
+    module.lambda_package
+  ]
+
+  filename         = local.zipfile
+  function_name    = var.name
+  handler          = "index.handler"
+  role             = aws_iam_role.this.arn
+  runtime          = "nodejs12.x"
+  source_code_hash = try(filebase64sha256(local.zipfile), null)
+
+  lifecycle {
+    ignore_changes = [
+      filename,
+      last_modified
+    ]
+  }
+
+  tags = merge(
+    var.tags,
+    {
+      "Name" = var.name
+    }
+  )
+}
+
+resource "aws_secretsmanager_secret" "privkey" {
+  name_prefix = "${var.name}-privkey-"
+  description = var.secret_description
+  tags = merge(
+    var.tags,
+    {
+      Name = "${var.name}-privkey"
+    }
+  )
+}
+
+resource "aws_secretsmanager_secret" "pubkey" {
+  name_prefix = "${var.name}-pubkey-"
+  description = var.secret_description
+  tags = merge(
+    var.tags,
+    {
+      Name = "${var.name}-pubkey"
+    }
+  )
```

Comment:
Don't we want to store the public key as an EC2 key pair? https://www.terraform.io/docs/providers/aws/r/key_pair.html

Reply:
I think it's best to do that in the parent module if desired. That's not necessarily why you're creating a keypair and you can't pull the pubkey from an ec2_key_pair as far as I can tell.

This isn't actually secret, but a secret is an easy way to manage it.

I did add a commit to add the pubkey as an output though.